### ISSUE TITLE
linter: plugin `typescript` is disabled by an empty `plugins` list

### ISSUE THREAD

**Issue by refparo** [oxc-project/oxc#22778]
### What version of Oxlint are you using?

1.67.0

### What command did you run?

`oxlint -c .oxlintrc.json`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "options": {
    "typeAware": true
  },
  "plugins": [], // remove this line and typescript/no-deprecated will work normally
  "rules": {
    "typescript/no-deprecated": "warn",
    // other enabled-by-default plugins, like `eslint`, are not disabled even if not listed in `plugins`
    "eqeqeq": ["warn", "always", { "null": "ignore" }],
  }
}
```

### What happened?

According to the document, the plugin `typescript` should be enabled by default, but an empty `plugins` list disables it. This behavior is not consistent with other enabled-by-default plugins.

For example, when the field `plugin` is present in .oxlintrc.json, typescript/no-deprecated stops working.

Minimal code that should trigger a warning:

```ts
/**
 * @deprecated
 */
function foo() {
}

foo();
```
